provider/core/src/datagen/mod.rs

```diff
@@ -93,11 +97,10 @@ pub trait DataExporter: Sync {
 pub trait ExportableProvider:
     IterableDynamicDataProvider<ExportMarker> + DynamicDataProvider<AnyMarker> + Sync
 {
-}
-
-impl<T> ExportableProvider for T where
-    T: IterableDynamicDataProvider<ExportMarker> + DynamicDataProvider<AnyMarker> + Sync
-{
+    /// Returns the source versions of the data.
```

Comment:
This breaks semver but I don't know how else I would get this data from the source provider. I really don't want to define a data key for this...

Reply:
Adding the function to the trait with a default impl doesn't break semver, but removing the blanket impl does. Can you use runtime specialization in the blanket impl to sniff for the known concrete impl that has a source_versions function? Well I guess that requires making `T: 'static` which is also semver. I lean toward just deleting the blanket impl at least to get the PR building and then we can escalate to the team.